src/frontend/src/catalog.rs

```diff
@@ -50,13 +51,44 @@ use table::TableRef;
 
 use crate::table::DistTable;
 
+// There are two sources for finding a table: the `local_catalog_manager` and the
+// `table_metadata_manager`.
+//
+// The `local_catalog_manager` is for storing tables that are often transparent, not saving any
+// real data. For example, our system tables, the `numbers` table and the "information_schema"
+// table.
+//
+// The `table_metadata_manager`, on the other hand, is for storing tables that are created by users,
+// obviously.
+//
+// For now, separating the two makes the code simpler, at least in the retrieval site. Now we have
+// `numbers` and `information_schema` system tables. Both have their special implementations. If we
+// put them with other ordinary tables that are created by users, we need to check the table name
+// to decide which `TableRef` to return. Like this:
+//
+// ```rust
+// match table_name {
+//   "numbers" => ... // return NumbersTable impl
+//   "information_schema" => ... // return InformationSchemaTable impl
+//   _ => .. // return DistTable impl
+// }
+// ```
+//
+// On the other hand, because we use `MemoryCatalogManager` for system tables, we can easily store
+// and retrieve the concrete implementation of the system tables by their names, no more "if-else"s.
+//
+// However, if the system table is designed to have more features in the future, we may revisit
+// the implementation here.
```

Comment:
- those are not doc comment
- add `,ignore` to the block language, otherwise it will become an uncompileable doc test

Reply:
intentionally not made them doc comments because they are not relevant to how to use this FrontendCatalogManager